test/unit/connectors/UpdateVerifiedEmailConnectorSpec.scala

```diff
@@ -0,0 +1,124 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.connectors
+
+import org.mockito.ArgumentMatchers.{any, eq => meq}
+import org.mockito.Mockito.{doNothing, reset, when}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, _}
+import org.scalatest.BeforeAndAfter
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatestplus.mockito.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import play.api.libs.json.Writes
+import play.api.test.Helpers._
+import uk.gov.hmrc.customs.emailfrontend.audit.Auditable
+import uk.gov.hmrc.customs.emailfrontend.config.AppConfig
+import uk.gov.hmrc.customs.emailfrontend.connectors.UpdateVerifiedEmailConnector
+import uk.gov.hmrc.customs.emailfrontend.connectors.http.responses.{BadRequest, NotFound, ServiceUnavailable, VerifiedEmailResponse}
+import uk.gov.hmrc.customs.emailfrontend.model.{RequestCommon, RequestDetail, UpdateVerifiedEmailRequest, UpdateVerifiedEmailResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class UpdateVerifiedEmailConnectorSpec extends PlaySpec
+  with ScalaFutures
+  with MockitoSugar
+  with BeforeAndAfter {
+
+  private val mockAuditable = mock[Auditable]
+  private val mockAppConfig = mock[AppConfig]
+  private val mockHttpClient = mock[HttpClient]
+
+  private val notFound = new NotFoundException("testMessage")
+  private val badRequestException = new BadRequestException("testMessage")
+  private val serviceUnavailableException = new ServiceUnavailableException("testMessage")
+
+  private val mockUpdateVerifiedEmailResponse = mock[UpdateVerifiedEmailResponse]
+
+  private val requestDetail = RequestDetail("idType", "idNumber", "test@email.com", "timestamp")
+  private val requestCommon = RequestCommon()
+
+  private val verifiedEmailResponse = VerifiedEmailResponse(mockUpdateVerifiedEmailResponse)
+
+  private val updateVerifiedEmailRequest = UpdateVerifiedEmailRequest(requestCommon, requestDetail)
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  val connector = new UpdateVerifiedEmailConnector(mockAppConfig, mockHttpClient, mockAuditable)
+
+  before {
+    reset(mockAuditable, mockAppConfig, mockHttpClient)
+    doNothing().when(mockAuditable).sendDataEvent(any(), any(), any(), any())(any[HeaderCarrier])
+    when(mockAppConfig.updateVerifiedEmailUrl).thenReturn("testUrl/update-verified-email")
+  }
+
+  "Calling updateVerifiedEmail" should {
+    "return Right with VerifiedEmailResponse when call was succesfull with OK" in {
```

Comment:
typo in `successful`? and do we need to be explicit about `Right` and `Left` in the test case description? can we not just say what is returned?

Reply:
thanks